LEGO1/mxdirectdraw.cpp

```diff
@@ -0,0 +1,968 @@
+        }
+    }
+    else
+    {
+        RECT rc;
+        DWORD dwStyle;
+
+        if (!m_bIsOnPrimaryDevice)
+        {
+            Error("Attempt made enter a windowed mode on a DirectDraw device that is not the primary display", E_FAIL);
+            return FALSE;
+        }
+
+        m_bIgnoreWM_SIZE = TRUE;
+        dwStyle = GetWindowLong(m_hWndMain, GWL_STYLE);
+        dwStyle &= ~(WS_POPUP | WS_CAPTION | WS_THICKFRAME | WS_OVERLAPPED);
+        dwStyle |= WS_CAPTION | WS_THICKFRAME | WS_OVERLAPPED;
+        SetWindowLong(m_hWndMain, GWL_STYLE, dwStyle);
+
+        temp_height = height;
+        temp_width = width;
+
+        SetRect(&rc, 0, 0, width - 1, height - 1);
+        AdjustWindowRectEx(
+            &rc,
+            GetWindowLong(m_hWndMain, GWL_STYLE),
+            GetMenu(m_hWndMain) != NULL,
+            GetWindowLong(m_hWndMain, GWL_EXSTYLE)
+        );
+        SetWindowPos(m_hWndMain, 0, 0, 0, rc.right - rc.left, rc.bottom - rc.top, SWP_NOMOVE | SWP_NOZORDER | SWP_NOACTIVATE);
+        SetWindowPos(m_hWndMain, HWND_NOTOPMOST, 0, 0, 0, 0, SWP_NOSIZE | SWP_NOMOVE | SWP_NOACTIVATE);
+        m_bIgnoreWM_SIZE = FALSE;
+
+        temp_bpp = bpp;
+    }
+
+    m_currentMode.width = temp_width;
+    m_currentMode.height = temp_height;
+    m_currentMode.bitsPerPixel = temp_bpp;
+
+    if (!DDCreateSurfaces())
+    {
+        return FALSE;
+    }
+
+    DDSURFACEDESC ddsd;
+
+    unk1();
+
+    if (!GetDDSurfaceDesc(&ddsd, m_pBackBuffer))
+    {
+        return FALSE;
+    }
+
+    if (ddsd.ddpfPixelFormat.dwFlags & DDPF_PALETTEINDEXED8)
+    {
+        m_bPrimaryPalettized = TRUE;
+    }
+    else
+    {
+        m_bPrimaryPalettized = FALSE;
+    }
+
+    if (m_bPrimaryPalettized)
+    {
+        result = m_pDirectDraw->CreatePalette(
+            DDPCAPS_8BIT | DDPCAPS_ALLOW256 | DDPCAPS_INITIALIZE, //0x4c
+            m_paletteEntries,
+            &m_pPalette,
+            NULL);
+        if (result != DD_OK)
+        {
+            Error("CreatePalette failed", result);
+            return 0;
+        }
+        result = m_pBackBuffer->SetPalette(m_pPalette); // TODO: add FIX_BUGS define and fix this
+        result = m_pFrontBuffer->SetPalette(m_pPalette);
+        if (result != DD_OK)
+        {
+            Error("SetPalette failed", result);
+            return FALSE;
+        }
+    }
+
+    if (m_bFullScreen)
+    {
+        return TRUE;
+    }
+    else
+    {
+        // create debug text only in windowed mode?
+        if (!CreateTextSurfaces())
+        {
+            return FALSE;
+        }
+        else
+        {
+            return TRUE;
+        }
+    }
+}
+
+// OFFSET: LEGO1 0x1009E830
+void MxDirectDraw::Error(const char* message, int error)
+{
+    if (!g_isInsideError)
+    {
+        g_isInsideError = TRUE;
+        Destroy();
+        if (m_pErrorHandler)
+        {
+            m_pErrorHandler(message, error, m_pErrorHandlerArg);
+        }
+        g_isInsideError = FALSE;
+    }
```

Comment:
I think this is a local static variable.
```suggestion
    static bool nsideError = FALSE;
    if (!insideError)
    {
        insideError = TRUE;
        Destroy();
        if (m_pErrorHandler)
        {
            m_pErrorHandler(message, error, m_pErrorHandlerArg);
        }
        insideError = FALSE;
    }
```